Create OneDollarDCAE

xiaopao2025 opened the pull request:
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):OneDollarDCAE


##### Twitter Link:


##### List of audit links if any:


##### Website Link:https://onedollardcae.xyz/


##### Logo (High resolution, will be shown with rounded borders):


##### Current TVL:little


##### Treasury Addresses (if the protocol has treasury)


##### Chain:Arbitrum


##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama):The OneDollarDCAE is an innovative decentralized finance (DeFi) solution that enables users to make efficient, automated investments in a tokenized ecosystem.


##### Token address and ticker if any:0x0c083720131a22F576815EE6a691C7Ed94E86Ab3


##### Category (full list at https://defillama.com/categories) *Please choose only one:


##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):TWAP
##### Implementation Details: Briefly describe how the oracle is integrated into your project: deploy won oracle to get twap price
##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:https://arbiscan.io/address/0xa87619defaa9b63f5d78ea69a4fbadea7341347e#code

##### forkedFrom (Does your project originate from another project):No


##### methodology (what is being counted as tvl, how is tvl being calculated):


##### Github org/user (Optional, if your code is open source, we can track activity):

llamabutler commented:
The adapter at projects/OneDollarDCAE exports TVL: 
        
 
 
    arbitrum                  79.69 k
    
    total                    79.69 k

g1nt0ki reviewed (changes requested):
.

llamabutler commented:
The adapter at projects/OneDollarDCAE exports TVL: 
        
 
 
    arbitrum                  10
    
    total                    9.852105908336238

g1nt0ki commented:
hi, I fixed the adapter, you need to have your own domain, we cant list with a vercel domain

xiaopao2025 commented:
Here's the new domain: dcaefront.9418666.xyz

g1nt0ki ***@***.***> 于 2025年2月19日周三 上午12:39写道：

> hi, I fixed the adapter, you need to have your own domain, we cant list
> with a vercel domain
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/13572#issuecomment-2666254622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4JISOG2GNR6VFAFIN5QMRT2QNO5DAVCNFSM6AAAAABXL42K4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRWGI2TINRSGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: g1nt0ki]*g1nt0ki* left a comment
> (DefiLlama/DefiLlama-Adapters#13572)
> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/13572#issuecomment-2666254622>
>
> hi, I fixed the adapter, you need to have your own domain, we cant list
> with a vercel domain
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/13572#issuecomment-2666254622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4JISOG2GNR6VFAFIN5QMRT2QNO5DAVCNFSM6AAAAABXL42K4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRWGI2TINRSGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

xiaopao2025 commented on projects/OneDollarDCAE/index.js:
```diff
@@ -0,0 +1,32 @@
+// OneDollarDCAE adapter for Arbitrum
+const USDC_TOKEN_ADDRESS = '0xaf88d065e77c8cC2239327C5EDb3A432268e5831'; // USDC on Arbitrum
+const WETH_TOKEN_ADDRESS = '0x82aF49447D8a07e3bd95BD0d56f35241523fBab1'; // WETH on Arbitrum
+const ONE_DOLLAR_DCAE_CONTRACT = '0x82aF49447D8a07e3bd95BD0d56f35241523fBab1'; // OneDollarDCAE contract address
```
Sorry, I forgot to modify the contract address: 0xA87619dEFaa9b63F5D78eA69a4fBAdEa7341347e. Thanks for your addition.

g1nt0ki commented:
again, this is not your own domain but a subdomain?

xiaopao2025 commented:
For my personal development project, I don't have much money, so I just use a new cheap domain name.

611314.xyz

g1nt0ki commented:
we accept any tvl over 0, sorry, but I think, it is reasonable to request for a proper domain name before listing?

xiaopao2025 commented:
I found a domain that was cheap and suitable, but only for a year, so maybe this is a compromise and hopefully I'll have enough money to maintain it in the future.

onedollardcae.xyz
